pkg/credentialprovider/gcp/metadata.go

```diff
@@ -148,18 +166,73 @@ func (g *dockerConfigUrlKeyProvider) Provide() credentialprovider.DockerConfig {
 	return credentialprovider.DockerConfig{}
 }
 
+func runWithBackoff(url string, f func() ([]byte, error)) ([]byte, error) {
```

Comment:
This never returns a non-nil error. It should either eventually timeout and return an error or you should document that it's a potentially infinite loop and just return the `[]byte`.